DM-17211 TAP Search Layout

tgoldina opened the pull request:
This is part 1 of DM-17211, which sets the layout and allows other people to work on the panel. 

https://jira.lsstcorp.org/browse/DM-17211

http://localhost:8080/firefly/firefly-dev.html?__action=layout.showDropDown&visible=true&view=TestTAPSearch

robyww commented:
>  Why is it under visualize.ui? Since there will be more than a few files, I would suggest putting them under ui.tap.

I think `ui.tap` is a great idea.

robyww commented on src/firefly/js/visualize/ui/TableSelectViewPanel.jsx:
```diff
@@ -49,9 +54,34 @@ export class TapSearchPanel extends PureComponent {
+                    <CreatableSelect
+                        options={TAP_SERVICE_OPTIONS}
+                        isClearable={true}
+                        onChange={(selectedOption) => {
```
I think long function inside the JSX make it hard to read.  If you need a closure function you declare it above or just make a static function that you use a closure to pass all the parameters to.

robyww commented on src/firefly/js/visualize/ui/TableSelectViewPanel.jsx:
```diff
@@ -65,35 +95,48 @@ export class TapSearchPanel extends PureComponent {
+                                    <TapSchemaBrowser serviceUrl={serviceUrl}/>
+                                    <div style={{position: 'absolute', top: 20, right: 20}}>
+                                        <button style={{padding: 3}}
+                                                onClick={ () => {
```
same here.

cwang2016 commented:
Review completes. It sets the good foundation for filing in the spatial/temporal search part.

tgoldina commented:
In the latest commit: 
- Moved TAP related panels to ui.tap
- Moved "Populate and edit ADQL" under the form
- Separated callback implementation from jsx
